GitUI/CommandsDialogs/SettingsDialog/ShellExtension/ShellExtensionManager.cs

```diff
@@ -0,0 +1,153 @@
+using System;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Diagnostics;
+using System.IO;
+using System.Reflection;
+using GitCommands;
+using Microsoft.Win32;
+
+namespace GitUI.CommandsDialogs.SettingsDialog.ShellExtension
+{
+    public static class ShellExtensionManager
+    {
+        /// <summary>
+        /// Checks if necessary for registration files were found
+        /// </summary>
+        /// <returns>false only if all necessary files were found, otherwise true</returns>
+        public static bool IsNotInstalled()
+        {
+            string path32 = FindFileInBinFolders(CommonLogic.GitExtensionsShellEx32Name);
+            string path64 = FindFileInBinFolders(CommonLogic.GitExtensionsShellEx64Name);
+            return string.IsNullOrEmpty(path32) || (Environment.Is64BitOperatingSystem && string.IsNullOrEmpty(path64));
+        }
+
+        /// <summary>
+        /// Checks if shell extension is properly registered in system
+        /// </summary>
+        /// <returns>false only if all checks are positive, otherwise true</returns>
+        public static bool IsNotRegistered()
+        {
+            return string.IsNullOrEmpty(CommonLogic.GetRegistryValue(Registry.LocalMachine, @"Software\Microsoft\Windows\CurrentVersion\Shell Extensions\Approved",
+                    "{3C16B20A-BA16-4156-916F-0A375ECFFE24}")) ||
+                string.IsNullOrEmpty(CommonLogic.GetRegistryValue(Registry.ClassesRoot,
+                    @"*\shellex\ContextMenuHandlers\GitExtensions2", null)) ||
```

Comment:
Keep on the same line. Reading this multiline statement with each sub-statement split across multiple lines is very difficult.
```suggestion
                string.IsNullOrEmpty(CommonLogic.GetRegistryValue(Registry.ClassesRoot, @"*\shellex\ContextMenuHandlers\GitExtensions2", null)) ||
```

Reply:
Specify the argument name for `null`s
```suggestion
                string.IsNullOrEmpty(CommonLogic.GetRegistryValue(Registry.ClassesRoot,
                    @"*\shellex\ContextMenuHandlers\GitExtensions2", key: null)) ||
```

Reply:
The code was extracted as is from ChecklistSettingsPage.cs so I thought it was preferred code style. Will fix it.

Reply:
As for the specifying argument name for null, I believe it's better to fix CommonLogic.GetRegistryValue